pkg/planner/core/exhaust_physical_plans.go

```diff
@@ -1377,10 +1387,105 @@ func (p *LogicalJoin) constructInnerIndexScanTask(
 	}
 	finalStats := ds.tableStats.ScaleByExpectCnt(rowCount)
 	is.addPushedDownSelection(cop, ds, tmpPath, finalStats)
-	t := cop.convertToRootTask(ds.SCtx())
-	reader := t.p
-	t.p = p.constructInnerByWrapper(wrapper, reader)
-	return t
+	return p.constructIndexJoinInnerSideTask(cop, ds, path, wrapper)
+}
+
+// construct the inner join task by inner child plan tree
+// The Logical include two parts: logicalplan->physicalplan, phsicaplan->task
+// Step1: whether agg can be pushed down to coprocessor
+//
+//	Step1.1: If the agg can be pushded down to coprocessor, we will build a copTask and attach the agg to the copTask
+//	There are two kinds of agg: stream agg and hash agg. Stream agg depends on some conditions, such as the group by cols
+//
+// Step2: build other inner plan node to task
+func (p *LogicalJoin) constructIndexJoinInnerSideTask(dsCopTask *copTask, ds *DataSource, path *util.AccessPath, wrapper *indexJoinInnerChildWrapper) task {
+	if len(wrapper.zippedChildren) == 0 {
+		t := dsCopTask.convertToRootTask(ds.SCtx())
+		return t
+	}
+	la, canPushAggToCop := wrapper.zippedChildren[len(wrapper.zippedChildren)-1].(*LogicalAggregation)
```

Comment:
Why can we assume there is only one Agg in `zippedChildren` and its index must be `len(zippedChildren)-1`?

Reply:
If there is an agg node at the bottom closest to the datasource, we need to consider whether it should be pushed down to tikv. If not, skip this part of the logic directly.